gating/pre_merge_test/run_deploy_mnaio.sh

```diff
@@ -0,0 +1,164 @@
+
+set -evu
+
+echo "Building a Multi Node AIO (MNAIO)"
+echo "+-------------------- MNAIO ENV VARS --------------------+"
+env
+echo "+-------------------- MNAIO ENV VARS --------------------+"
+
+## Vars ----------------------------------------------------------------------
+
+# These vars are set by the CI environment, but are given defaults
+# here to cater for situations where someone is executing the test
+# outside of the CI environment.
+export RE_JOB_NAME="${RE_JOB_NAME:-}"
+export RE_JOB_IMAGE="${RE_JOB_IMAGE:-}"
+export RE_JOB_BRANCH="${RE_JOB_BRANCH:-newton}"
+export RE_JOB_SCENARIO="${RE_JOB_SCENARIO:-swift}"
+export RE_JOB_ACTION="${RE_JOB_ACTION:-deploy}"
+export RE_JOB_FLAVOR="${RE_JOB_FLAVOR:-}"
+export RE_JOB_TRIGGER="${RE_JOB_TRIGGER:-PR}"
+export RE_HOOK_ARTIFACT_DIR="${RE_HOOK_ARTIFACT_DIR:-/tmp/artifacts}"
+export RE_HOOK_RESULT_DIR="${RE_HOOK_RESULT_DIR:-/tmp/results}"
+
+## OSA MNAIO Vars
+export PARTITION_HOST="true"
+export NETWORK_BASE="172.29"
+export DNS_NAMESERVER="8.8.8.8"
+export OVERRIDE_SOURCES="true"
+export DEVICE_NAME="vda"
+export DEFAULT_NETWORK="eth0"
+export DEFAULT_IMAGE="ubuntu-16.04-amd64"
+export DEFAULT_KERNEL="linux-image-generic"
+export SETUP_HOST="true"
+export SETUP_VIRSH_NET="true"
+export VM_IMAGE_CREATE="true"
+export DEPLOY_OSA="true"
+export PRE_CONFIG_OSA="true"
+export RUN_OSA="false"
+export CONFIGURE_OPENSTACK="false"
+export DATA_DISK_DEVICE="sdb"
+export CONFIG_PREROUTING="false"
+export OSA_PORTS="6080 6082 443 80 8443"
+export RPC_BRANCH="${RE_JOB_BRANCH}"
+export DEFAULT_MIRROR_HOSTNAME=mirror.rackspace.com
+export DEFAULT_MIRROR_DIR=/ubuntu
+export INFRA_VM_SERVER_RAM=16384
+
+# ssh command used to execute tests on infra1
+export MNAIO_SSH="ssh -ttt -oStrictHostKeyChecking=no root@infra1"
+
+## Functions -----------------------------------------------------------------
+source /opt/rpc-openstack/scripts/functions.sh
+
+## Main --------------------------------------------------------------------
+
+# capture all RE_ variables
+env | grep RE_ | sed 's/^/export /' > /opt/rpc-openstack/RE_ENV
+
+# checkout openstack-ansible-ops
+if [ ! -d "/opt/openstack-ansible-ops" ]; then
+  git clone --recursive https://github.com/openstack/openstack-ansible-ops /opt/openstack-ansible-ops
+else
+  pushd /opt/openstack-ansible-ops
+    git fetch --all
+  popd
+fi
+
+# set OSA branch
+pushd /opt/rpc-openstack
+  OSA_COMMIT=`git submodule status openstack-ansible | egrep --only-matching '[a-f0-9]{40}'`
+  export OSA_BRANCH=${OSA_COMMIT}
+popd
+
+# apply various modifications for mnaio
+pushd /opt/openstack-ansible-ops/multi-node-aio
+  # By default the MNAIO deploys metering services, so we override
+  # osa_enable_meter to prevent those services from being deployed.
+  sed -i 's/osa_enable_meter: true/osa_enable_meter: false/' playbooks/group_vars/all.yml
+  # bump up cryptography version to avoid exception detailed in RLM-1104
+  pip install cryptography==1.5 --upgrade
```

Comment:
The only issue here is that there's no guarantee that pip is installed at this stage.  I just ran it on a fresh (manual) deploy and it failed because no pip.

Reply:
As discussed, this can go in until rcbops/rpc-gating#591 merges, at which point it shouldn't be necessary and can be reverted.